src/wh_client.c

```diff
@@ -733,6 +733,203 @@ int wh_Client_KeyErase(whClientContext* c, whNvmId keyId)
     return ret;
 }
 
+int wh_Client_CounterInitRequest(whClientContext* c, whNvmId keyId,
+    uint32_t counter)
+{
+    whPacket* packet;
+    if (c == NULL || keyId == WOLFHSM_KEYID_ERASED)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    /* set keyId and initial value */
+    packet->counterInitReq.keyId = keyId;
+    packet->counterInitReq.counter = counter;
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_COUNTER, WH_COUNTER_INIT,
+        WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->counterInitReq),
+        (uint8_t*)packet);
+}
+
+int wh_Client_CounterInitResponse(whClientContext* c, uint32_t* counter)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    whPacket* packet;
+    if (c == NULL || counter == NULL)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+        else
+            *counter = packet->counterInitRes.counter;
+    }
+    return ret;
+}
+
+int wh_Client_CounterInit(whClientContext* c, whNvmId keyId, uint32_t* counter)
```

Comment:
Consider splitting the input counter and the output counter parameters and allow null for output counter.